altos-rust/altos-core/src/task/control.rs

```diff
@@ -63,6 +69,13 @@ impl Priority {
     (0..NUM_PRIORITIES)
   }
 
+
+  /// Returns a box of the range of values of Priority minus the one passed in.
+  pub fn all_except<'a>(priority:Priority) -> Box<Iterator<Item=usize> + 'a> {
+    let all = Priority::all();
+    Box::new(all.filter(move |&x| x != priority as usize))
```

Comment:
`  pub fn all_but(n: u8) -> ::core::iter::Chain<core::ops::Range<u8>,
                                               core::ops::Range<u8>> {
    (0..n).chain(n+1..P::I as u8)
  }
}
`
Feels a little hacky, but this could give us what we want too.